File: Admin/app/src/main/java/com/example/adminbookinghotel/Login/FogotPassword.java
package com.example.adminbookinghotel.Login;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Patterns;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.adminbookinghotel.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

public class FogotPassword extends AppCompatActivity {

    private EditText edt_email;
    private FirebaseAuth auth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fogot_password);

        edt_email = findViewById(R.id.edt_fogotPass);
        Button button = findViewById(R.id.btn_XacNhan);
        auth = FirebaseAuth.getInstance();
        Button btn_back = findViewById(R.id.btn_Back);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ForgotPassword();
            }
        });
        btn_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getApplicationContext(), SignIn.class));
                finish();
            }
        });
    }

    private boolean Check(String name) {
        return !TextUtils.isEmpty(name);
    }

    private void ForgotPassword() {
        String strEmail = edt_email.getText().toString().trim();
        if (!Check(strEmail)) {
            edt_email.setError("Please enter Email!");
            edt_email.requestFocus();
        } else if (!Patterns.EMAIL_ADDRESS.matcher(edt_email.getText().toString().trim()).matches()) {
            edt_email.setError("Email wrong format!");
            edt_email.requestFocus();
        } else {
            auth.sendPasswordResetEmail(strEmail).addOnCompleteListener(new OnCompleteListener<Void>() {
                @Override
                public void onComplete(@NonNull Task<Void> task) {
                    if (task.isSuccessful()) {
                        startActivity(new Intent(FogotPassword.this, SignIn.class));
                        ShowToast("Check mail!");
                        finish();
                    } else {
                        ShowToast("Email not exist!");
                    }
                }
            });
        }
    }

    private void ShowToast(String messenger) {
        Toast.makeText(this, messenger, Toast.LENGTH_SHORT).show();
    }
}

File: User/app/src/main/java/com/example/bookinghotel/signIn_Up/FogotPassword.kt
package com.example.bookinghotel.signIn_Up

import android.content.Intent
import android.os.Bundle
import android.util.Patterns
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.bookinghotel.databinding.FogotPasswordBinding
import com.google.firebase.auth.FirebaseAuth

class FogotPassword : AppCompatActivity() {

    private lateinit var binding: FogotPasswordBinding
    private lateinit var auth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        binding = FogotPasswordBinding.inflate(layoutInflater)
        setContentView(binding.root)
        super.onCreate(savedInstanceState)

        auth = FirebaseAuth.getInstance()

        binding.btnBack.setOnClickListener() {
            onBackPressed()
        }

        binding.btnXacNhan.setOnClickListener {
            fogotPassword()
        }

    }

    private fun fogotPassword() {
        val strEmail = binding.edtFogot.text.toString().trim()
        if (strEmail.isEmpty()) {
            binding.edtFogot.error = "Please enter email !"
            binding.edtFogot.requestFocus()
            return
        } else if (!Patterns.EMAIL_ADDRESS.matcher(strEmail).matches()) {
            binding.edtFogot.error = "Email invalid"
            binding.edtFogot.requestFocus()
            return
        } else {
            auth.sendPasswordResetEmail(strEmail).addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    showToast("Please Check email")
                    startActivity(Intent(this, SignIn::class.java))
                    finish()
                } else {
                    showToast("Email not exist")
                }
            }
        }
    }

    private fun showToast(mess: String) {
        Toast.makeText(applicationContext, mess, Toast.LENGTH_SHORT).show()
    }
}